Routes/doctor.js

```diff
@@ -0,0 +1,437 @@
+ *                     type: string
+ *          PrescriptionMedicine:
+ *             type: object
+ *             properties:
+ *                 prescriptionId:
+ *                     type: string
+ *                 medicineId:
+ *                     type: string
+ *          Medicine:
+ *             type: object
+ *             properties:
+ *                 medicineName:
+ *                     type: string
+ *                 medicineType:
+ *                     type: string
+ *                 medicineCost:
+ *                     type: number
+ *                 medicineDescription:
+ *                     type: string
+ *          Hospital:
+ *             type: object
+ *             properties:
+ *                 hospitalName:
+ *                     type: string
+ *                 hospitalAddress:
+ *                     type: object
+ *                     properties:
+ *                         address_line:
+ *                             type: string 
+ *                         City:
+ *                             type: string
+ *                         Postal_Code:
+ *                             type: number
+ *                         Country:
+ *                             type: string
+ *                 noOfRooms:
+ *                     type: number
+ *                 roomCharge:
+ *                     type: number
+ *          bill:
+ *             type: object
+ *             properties:
+ *                 prescriptionId:
+ *                      type: string
+ *                 doctorCharge:
+ *                      type: number
+ *                 noOfDays:
+ *                      type: number
+ *                 operationCharge:
+ *                      type: number                     
+ *          
+ *             
+ */
+
+/**
+ * @swagger
+ * /doctor/newDoctor:
+ *  post:
+ *      summary: Registering a new Doctor  
+ *      description: Add the new doctor data in database
+ *      requestBody:
+ *          required: true
+ *          content:
+ *              application/json:
+ *                  schema:
+ *                      $ref: '#/components/schemas/Doctor'
+ *      responses:
+ *          200:
+ *              description: Added Successfully
+ *              
+ */ 
+
+DoctorRoute.post('/newDoctor',async (req,res) => {
+    const hospital = await Hospital.findById(req.body.hospitalId)
+    if(hospital != null || hospital != undefined){
+
+        const doctor = new Doctors({
+            doctorName: req.body.doctorName,
+            doctorEmail: req.body.doctorEmail,
+            doctorPassword: await bcrypt.hash(req.body.doctorPassword,10),
+            doctorDepartment: req.body.doctorDepartment,
+            hospitalId: req.body.hospitalId,
+            hospitalName: hospital.hospitalName
+            
+        })
+        const newDoctor = await doctor.save()
+        res.json(newDoctor);  
+    }
+    else{
+        res.json({result: "incorrect hospital id try again"})
+    }
+    
+})
```

Comment:
The logic of an API endpoint shouldn't be written in the route file, instead, it should be written in a separate file that is the controller file, this is how the folder structure will look like for a very basic API system.

```
src
  |_ controllers
    |_ example.controller.js
  |_ routes
    |_ example.routes.js
  |_ models  
    |_ example.model.js
```

Reply:
Followed and edited the folder and file structure [PR](https://github.com/Sankalp-Sachapara/HospitalPractice/pull/2/files/893ce87fe4fe8a5be2c037597a3a6cfbdffc56ac#r1045072926)